app/src/main/java/com/sun/heartrate/ui/heartbeat/HeartbeatFragment.kt

```diff
@@ -1,13 +1,57 @@
 package com.sun.heartrate.ui.heartbeat
 
+import android.Manifest
+import android.annotation.SuppressLint
+import android.annotation.TargetApi
+import android.content.Context
+import android.content.pm.PackageManager
+import android.graphics.ImageFormat
+import android.hardware.camera2.*
+import android.media.ImageReader
+import android.os.Build
 import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
 import android.view.LayoutInflater
+import android.view.Surface
 import android.view.View
 import android.view.ViewGroup
+import androidx.core.app.ActivityCompat
 import androidx.fragment.app.Fragment
 import com.sun.heartrate.R
+import com.sun.heartrate.ui.heartbeat.callback.OnImageAvailableListener
+import com.sun.heartrate.ui.heartbeat.callback.SessionStateCallback
+import com.sun.heartrate.ui.heartbeat.callback.StateCallback
+import kotlinx.android.synthetic.main.fragment_heartbeat.*
 
-class HeartbeatFragment : Fragment() {
+@SuppressLint("NewApi")
+@TargetApi(Build.VERSION_CODES.KITKAT)
+class HeartbeatFragment : Fragment(), HeartbeatContract.View, View.OnClickListener {
+    
+    private val heartbeatPresenter: HeartbeatPresenter by lazy {
```

Comment:
```suggestion
    private val heartbeatPresenter: HeartbeaContract.tPresenter by lazy {
```
Các class giao tiếp vs nhau qua interface